### PR TITLE
BMP-173: User Preferences & Recommendations

### PR DESCRIPTION
**Pull request by edcav18**
## Summary
This PR extends the Groq service with new functionality for **user preferences extraction** and **event recommendations**, along with schema updates and unit tests.

## Changes by Greg (BMP-162)
- Expanded the JSON schema with new models:
  - `UserPreferences`
  - `UserPreferencesResponse`
  - `Recommendation`
  - `RecommendationsResponse`
- Implemented the **`get_user_preferences`** function.
- Added the **`/get_user_preferences`** endpoint.
- Wrote a **unit test** for `get_user_preferences`.

## Changes by Me (BMP-173)
- Implemented the **`create_recommendations`** function.
- Updated schemas to support recommendations.
- Added the **`/create_recommendations`** endpoint.
- Wrote a **unit test** for `create_recommendations`.

## Technical Notes
- Both features rely on Groq prompts to output valid JSON that conforms to the updated schemas.
- Unit tests are written with `pytest` and use a `FakeAsyncClient` for mocking responses.
- This work lays the foundation for integrating AI-powered event recommendations into the backend.

## Next Steps
- Expand test coverage for edge cases (invalid JSON, empty preferences, no candidate events).
- Integrate recommendations with frontend search results once endpoints are stable.

### CODE REVIEW

File: src/providers/groq/groq_service.py [hunk 1]
```diff
@@ -138,6 +170,95 @@ async def get_summary(self) -> SummaryResponse:
...
+            return UserPreferencesResponse(**data.model_dump())
+        except ValidationError:
+            logger.warning("Schema validation failed for user preferences output: %s", content)
+            return UserPreferencesResponse(
+                genres=[],
+                artists=[],
+                locations=[]
+            )
+
+    async def create_recommendations(self, user_preferences: UserPreferences, events: list) -> RecommendationsResponse:
```
Comment (samuelakerman) [inline src/providers/groq/groq_service.py:219]: both parameters should be of a defined schema. UserPreferences is good. But "list" for events is too ambiguous, we should define an object Event so that we pass a list of Event.

File: src/backend/app/main.py
Comment (samuelakerman) [inline src/backend/app/main.py:1]: I was expecting to see a method here that would put the logic together to call core/groq_client.py:create_recommendations()
ie, 1) get user preferences and 2) a list of events so that the new functionality can be called from the backend.

**Review by samuelakerman — CHANGES_REQUESTED** [review]
